src/engine/BMBtnSkill.php

```diff
@@ -20,4 +20,34 @@ class BMBtnSkill {
     public static function get_description() {
         return NULL;
     }
+
+    /**
+     * Complete description of skill, packaged for front end
+     *
+     * @param string $skill
+     * @return array|NULL
+     */
+    public static function describe($skill, $includeEmpty = TRUE) {
+        $skillClass = "BMBtnSkill$skill";
+        $description = $skillClass::get_description();
+
+        if (!$includeEmpty) {
+            if (empty($description)) {
+                return NULL;
+            }
+
+            $skillInstance = new $skillClass;
+            if ($skillInstance instanceof BMBtnSkillArtificialReenable) {
+                return NULL;
+            }
+        }
+
+        $skillDescription = array(
+            'code' => $skill,
```

Comment:
Yeah, okay, now i understand what you're doing here that's leading to the doubled "Giant", and it's definitely a bug.  I guess it happens that the UI doesn't assume codes are single-character strings found in recipes, but there's no reason someone couldn't make that assumption in using API output, so i don't think it makes sense to break that.

I think your easiest solution is to leave 'code' empty (or not pass it), and have the UI not display the parens if there is no code.  If you wanted to pass a new parameter (e.g. 'type' => 'die' or 'type' => 'button'), you could, but that would be a lot more test updating, so i don't think you need to.  I think overloading 'code' is a bug.